### PR TITLE
 [12.0][3697][ADD] product_standard_price_restriction

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3697](https://www.quartile.co/web#id=3697&cids=3&model=project.task&view_type=form)

### CODE REVIEW

File: product_standard_price_restriction/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (yostashiro) [inline product_standard_price_restriction/models/product_template.py]:
```suggestion
```

File: product_standard_price_restriction/security/product_security.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="group_product_manager" model="res.groups">
```
Comment (yostashiro) [inline product_standard_price_restriction/security/product_security.xml]: @rinaldifirdaus Do you think we should add this group this time, according to the requirement discussed with the client?

File: product_standard_price_restriction/readme/DESCRIPTION.rst
Comment (yostashiro) [inline product_standard_price_restriction/readme/DESCRIPTION.rst:1]: @rinaldifirdaus Please add the business background.

File: product_standard_price_restriction/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    standard_price = fields.Float(track_visibility='onchange')
+    update_cost = fields.Boolean()
+
+    def write(self, vals):
+        if "standard_price" in vals:
+            if not vals.get('update_cost', self.update_cost):
+                raise UserError(
+                    _(
+                        "You are about to update the cost. This field is "
+                        "auto-calculated by Odoo and is an 'Weighted Average Value' of "
```
Comment (yostashiro) [inline product_standard_price_restriction/models/product_template.py]: It is not good to hardcode a message like this. Can we make the message more generic, or put it in a configuration field (upon adding a field perhaps in res.company)?

**Review by rinaldifirdaus — APPROVED** [review]
Functional test: Working as Expected!

File: product_standard_price_restriction/models/product_template.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    standard_price = fields.Float(track_visibility='onchange')
+    update_cost = fields.Boolean()
```
Comment (yostashiro) [inline product_standard_price_restriction/models/product_template.py]: Add help.

**Review by yostashiro — APPROVED** [review]

**Pull request merged**